### PR TITLE
clean tmp file after upload

### PR DESCRIPTION
**Pull request by fengxx**
CopyArchiveToContainerCmdImpl kept temp file on disk, need remove it after upload

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/848)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java [hunk 1]
```diff
@@ -131,11 +125,18 @@ public Void exec() throws NotFoundException {
                         "Only one of host resource or tar input stream should be defined to perform the copy, not both");
             }
             // We compress the given path, call exec so that the stream is consumed and then close it our self
-            try (InputStream uploadStream = buildUploadStream(this.hostResource, this.dirChildrenOnly)) {
-                this.tarInputStream = uploadStream;
+            Path toUpload = null;
+            try {
+                toUpload = Files.createTempFile("docker-java", ".tar.gz");
+                CompressArchiveUtil.tar(Paths.get(hostResource), toUpload, true, dirChildrenOnly);
+                this.tarInputStream = Files.newInputStream(toUpload);
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: should this be wrapped with try-with-resources or it closed in somewhere else place?

Comment (fengxx) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: Fixed

File: src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java [hunk 2]
```diff
@@ -131,11 +125,20 @@ public Void exec() throws NotFoundException {
...
             }
             // We compress the given path, call exec so that the stream is consumed and then close it our self
-            try (InputStream uploadStream = buildUploadStream(this.hostResource, this.dirChildrenOnly)) {
+            Path toUpload;
+            try {
+                toUpload = Files.createTempFile("docker-java", ".tar.gz");
+                CompressArchiveUtil.tar(Paths.get(hostResource), toUpload, true, dirChildrenOnly);
+            } catch (IOException createFileIOException) {
+                throw new DockerClientException("Unable to perform tar on host resource " + this.hostResource, createFileIOException);
+            }
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: should it have finally also here with toUpload remove?

File: src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java [hunk 3]
```diff
@@ -131,11 +125,20 @@ public Void exec() throws NotFoundException {
...
+            try (InputStream uploadStream = Files.newInputStream(toUpload)) {
                 this.tarInputStream = uploadStream;
                 return super.exec();
             } catch (IOException e) {
-                throw new DockerClientException("Unable to perform tar on host resource " + this.hostResource, e);
+                throw new DockerClientException("Unable to read temp file " + toUpload.toFile().getAbsolutePath(), e);
+            } finally {
+                toUpload.toFile().delete();
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: please place comment `// remove tmp dir`

Comment (fengxx) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: added

File: src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java [hunk 4]
```diff
@@ -131,11 +125,25 @@ public Void exec() throws NotFoundException {
...
+                toUpload = Files.createTempFile("docker-java", ".tar.gz");
+                CompressArchiveUtil.tar(Paths.get(hostResource), toUpload, true, dirChildrenOnly);
+            } catch (IOException createFileIOException) {
+                if (toUpload != null) {
+                    // remove tmp docker-javaxxx.tar.gz
+                    toUpload.toFile().delete();
+                }
+                throw new DockerClientException("Unable to perform tar on host resource " + this.hostResource, createFileIOException);
+            }
+            try (InputStream uploadStream = Files.newInputStream(toUpload)) {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: add new line before for better readability and squash commits (to keep contribution in GH)

File: src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java [hunk 5]
```diff
@@ -131,11 +125,27 @@ public Void exec() throws NotFoundException {
...
+                throw new DockerClientException(
+                        "Unable to perform tar on host resource " + this.hostResource, createFileIOException);
+            }
+            try (InputStream uploadStream = Files.newInputStream(toUpload)) {
                 this.tarInputStream = uploadStream;
                 return super.exec();
             } catch (IOException e) {
-                throw new DockerClientException("Unable to perform tar on host resource " + this.hostResource, e);
+                throw new DockerClientException(
+                        "Unable to read temp file " + toUpload.toFile().getAbsolutePath(), e);
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: 120..140 width is acceptable :)

Comment (fengxx) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: thanks for patience,  formatted code use 140

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java]: ok

File: .idea/codeStyleSettings.xml [hunk 6]
```diff
@@ -5,6 +5,7 @@
       <value>
         <option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="9999" />
         <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="9999" />
+        <option name="RIGHT_MARGIN" value="140" />
```
Comment (KostyaSha) [inline .idea/codeStyleSettings.xml]: https://github.com/docker-java/docker-java/blob/master/src/test/resources/checkstyle/checkstyle-config.xml#L91-L92
thanks. Added file only some time before, not fully filled.

**Pull request merged**